api/src/main/java/org/apache/iceberg/SnapshotRef.java

```diff
@@ -0,0 +1,186 @@
+  }
+
+  public Long maxSnapshotAgeMs() {
+    return maxSnapshotAgeMs;
+  }
+
+  public Long maxRefAgeMs() {
+    return maxRefAgeMs;
+  }
+
+  public static Builder tagBuilder(long snapshotId) {
+    return builderFor(snapshotId, SnapshotRefType.TAG);
+  }
+
+  public static Builder branchBuilder(long snapshotId) {
+    return builderFor(snapshotId, SnapshotRefType.BRANCH);
+  }
+
+  public static Builder builderFrom(SnapshotRef ref) {
+    return new Builder(ref.type(), ref.snapshotId())
+        .minSnapshotsToKeep(ref.minSnapshotsToKeep())
+        .maxSnapshotAgeMs(ref.maxSnapshotAgeMs())
+        .maxRefAgeMs(ref.maxRefAgeMs());
+  }
+
+  /**
+   * Creates a ref builder from the given ref and its properties but the ref will now point to the given snapshotId.
+   * @param ref Ref to build from
+   * @param snapshotId snapshotID to use.
+   * @return ref builder with the same retention properties as given ref, but the ref will point to the passed in id
+   */
+  public static Builder builderFrom(SnapshotRef ref, long snapshotId) {
+    return new Builder(ref.type(), snapshotId)
+            .minSnapshotsToKeep(ref.minSnapshotsToKeep())
+            .maxSnapshotAgeMs(ref.maxSnapshotAgeMs())
+            .maxRefAgeMs(ref.maxRefAgeMs());
+  }
+
+
+  public static Builder builderFor(long snapshotId, SnapshotRefType type) {
+    return new Builder(type, snapshotId);
+  }
+
+  public static class Builder {
+
+    private final SnapshotRefType type;
+    private final long snapshotId;
+
+    private Integer minSnapshotsToKeep;
+    private Long maxSnapshotAgeMs;
+    private Long maxRefAgeMs;
+
+    Builder(SnapshotRefType type, long snapshotId) {
+      Preconditions.checkArgument(type != null, "Snapshot reference type must not be null");
+      this.type = type;
+      this.snapshotId = snapshotId;
+    }
+
+    public Builder minSnapshotsToKeep(Integer value) {
+      this.minSnapshotsToKeep = value;
+      return this;
+    }
+
+    public Builder maxSnapshotAgeMs(Long value) {
+      this.maxSnapshotAgeMs = value;
+      return this;
+    }
+
+    public Builder maxRefAgeMs(Long value) {
+      this.maxRefAgeMs = value;
+      return this;
+    }
+
+    public SnapshotRef build() {
+      if (type.equals(SnapshotRefType.TAG)) {
+        ValidationException.check(minSnapshotsToKeep == null,
+            "Tags do not support setting minSnapshotsToKeep");
+        ValidationException.check(maxSnapshotAgeMs == null,
+            "Tags do not support setting maxSnapshotAgeMs");
+      } else {
+        ValidationException.check(minSnapshotsToKeep == null || minSnapshotsToKeep > 0,
+                "Min snapshots to keep must be greater than 0");
+        ValidationException.check(maxSnapshotAgeMs == null || maxSnapshotAgeMs > 0,
+                "Max snapshot age must be greater than 0");
+      }
+
+      if (maxRefAgeMs != null) {
+        ValidationException.check(maxRefAgeMs > 0, "Max reference age must be greater than 0");
```

Comment:
This can be rewritten to avoid the unnecessary `if`:

```java
ValidationException.check(maxRefAgeMs == null || maxRefAgeMs > 0, "Max reference age must be greater than 0");
```